nodes/curve/deconstruct_curve.py

```diff
@@ -0,0 +1,112 @@
+import numpy as np
+
+from mathutils import Matrix, Vector
+import bpy
+from bpy.props import FloatProperty, EnumProperty, BoolProperty, IntProperty
+
+from sverchok.node_tree import SverchCustomTreeNode, throttled
+from sverchok.data_structure import updateNode, zip_long_repeat, ensure_nesting_level, get_data_nesting_level
+from sverchok.utils.curve import SvCurve
+from sverchok.utils.curve.nurbs import SvNurbsCurve
+
+class SvDeconstructCurveNode(bpy.types.Node, SverchCustomTreeNode):
+    """
+    Triggers: Deconstruct Curve
+    Tooltip: Output degree, control points, weights, knot vector of the curve (when they are defined)
+    """
+    bl_idname = 'SvDeconstructCurveNode'
+    bl_label = 'Deconstruct Curve'
+    bl_icon = 'OUTLINER_OB_EMPTY'
+    sv_icon = 'SV_DECONSTRUCT_CURVE'
+
+    def sv_init(self, context):
+        self.inputs.new('SvCurveSocket', "Curve")
+        self.outputs.new('SvStringsSocket', "Degree")
+        self.outputs.new('SvStringsSocket', "KnotVector")
+        self.outputs.new('SvVerticesSocket', "ControlPoints")
+        self.outputs.new('SvStringsSocket', "Edges")
+        self.outputs.new('SvStringsSocket', "Weights")
+
+    def deconstruct(self, curve):
```

Comment:
This method does not look like belong to the node class. It even does not use self argument. Most likely the proper place would be: `curve.deconstruct()`